src-built-in/components/authentication/oauthResponse.html

```diff
@@ -0,0 +1,108 @@
+<!DOCTYPE html>
+<html>
+<!--
+    /*!
+    * Copyright 2017 by ChartIQ, Inc.
+    * All rights reserved.
+    */
+	OAUTH Authentication example. This example uses Google as a OpenID Provider (OP).
+
+-->
+
+<head>
+	<meta charset="utf-8">
+	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+	<title>Oauth Authentification</title>
+	<meta name="description" content="">
+	<meta name="viewport" content="width=device-width, initial-scale=1">
+	<script>
+		function closeThisWindow(){
+			// When we close this window, if the user isn't logged in then it must mean we
+			// were using OAUTH to log in, and it failed, so shut down the entire app. Otherwise
+			// we were just using it to authenticate a client, in which case we just want to close
+			// this window but keep Finsemble running.
+			FSBL.Clients.AuthenticationClient.getCurrentCredentials(function(err, credentials){
+				if(!credentials){
+					FSBL.shutdownApplication();
+				}else{
+					fin.desktop.Window.getCurrent().close(true);
```

Comment:
Would you replace this with a FSBL call? I think FSBL.Clients.WindowClient.getCurrent().close(true) should do it.

Reply:
@NinjaManatee I reworked this a bit but with regards to SalesForce, the page for that in the sales demo is very different: https://github.com/ChartIQ/finsemble-sales-demo/blob/master/src/components/salesforce/oauth2/authed.html

I might be missing something.